core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -189,13 +188,12 @@ private[spark] class HighlyCompressedMapStatus private (
     emptyBlocks.readExternal(in)
     avgSize = in.readLong()
     val count = in.readInt()
-    val hugeBlockSizesArray = mutable.ArrayBuffer[Tuple2[Int, Byte]]()
+    hugeBlockSizes = mutable.Map.empty[Int, Byte]
     (0 until count).foreach { _ =>
       val block = in.readInt()
       val size = in.readByte()
-      hugeBlockSizesArray += Tuple2(block, size)
+      hugeBlockSizes.asInstanceOf[mutable.Map[Int, Byte]].update(block, size)
```

Comment:
@srowen I used more generic map type in HighlyCompressedMapStatus, so cast was to make update operations on it possible. However, it does not seem necessary because HighlyCompressedMapStatus fields are not exposed outside. So we can simply change type of the map right there.

Reply:
Yes, it is a mutable map and used as a mutable map. Its type must reflect that.